replay/models/dt4rec/gpt1.py

```diff
@@ -0,0 +1,409 @@
+# pylint: disable=invalid-name
+
+import logging
+import math
+
+import torch
```

Comment:
torch should be optional dependency. Use if TORCH_AVAILABLE in the similar manner as with PyPspark here for example https://github.com/sb-ai-lab/RePlay/blob/main/replay/models/knn.py#L9